### DISCUSSION TITLE
CoApp still not recognised

### DISCUSSION THREAD

**Discussion by hector2o** [aclap-dev/video-downloadhelper#27] [Help and Q&A] [answered]
### Your question

A clear description of the problem: Downloadhelper in settings says "Coapp not installed". After making a validation it says "No such native application net.downloadhelper.coapp. Click here to troubleshoot your issue." -> Then landing here in github site "https://github.com/aclap-dev/vdhcoapp/blob/master/doc/Troubleshooting.md" I read all advices in wiki and similar discussions, but nothing helped. 

### Addon & CoApp version and details

To your wished questions here all my details:
The browser you're using (be as specific as possible) -> Firefox 122.0 (64-Bit) for Ubuntu, canonical 1.0
    OS (Linux/Windows/Mac) & OS version -> Ubuntu 20.04.5 LTS (Linux x86-64)
    OS 64bits or 32bits -> 64bit
    Link to the coapp you installed -> https://github.com/aclap-dev/vdhcoapp/releases/download/v2.0.10/vdhcoapp-2.0.10-linux-x86_64.deb 
    On Linux, your sandbox mechanism (if any): flatpak/snap/… -> snap and flatpak installed additionally
    On Linux, the output of /opt/vdhcoapp/vdhcoapp --info ->
    {
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/mac7pro",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"

in the coapp is installed in opt/vdhcoapp:
/opt/vdhcoapp$ ll
insgesamt 141388
drwxr-xr-x 2  502 staff     4096 Feb  3 18:53 ./
drwxr-xr-x 4 root root      4096 Jan 13 11:28 ../
-rwxr-xr-x 1  502 staff 48653608 Dez 14 05:59 ffmpeg*
-rwxr-xr-x 1  502 staff 48539368 Dez 14 05:59 ffprobe*
-rw-r--r-- 1  502 staff    15220 Dez 14 05:59 LICENSE.txt
-rw-r--r-- 1  502 staff     4955 Dez 14 05:59 README.md
-rwxr-xr-x 1  502 staff 47515088 Dez 14 05:59 vdhcoapp*
-rwxr-xr-x 1  502 staff    25778 Dez 14 05:59 xdg-open*

Last Update OS (Ubuntu): Today, 3rd of February 2024, around 10 am

**Comment by GrampaWildWilly** [comment]
You left out a couple of bits of information.  As a model, here's my setup: Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0 64-bit, licensed VDH 8.2.0.18a1 beta, CoApp 2.0.14.  You are definitely behind on the CoApp.  I don't know what you are using for the VDH browser extension since you left out that tidbit.

So my first idea is that you should get on the current levels of the software.

My second idea is for you to read this one:

https://github.com/aclap-dev/video-downloadhelper/discussions/14

**Reply by hector2o** [reply]
Sorry Grampa, you are right, I've forgotten the VDH Version. To your first idea -> now I updated all I could today:
Firefox for Ubuntu -> 122.0.1 (Today)
VDH -> 8.2.0.19a1 (beta) (Today)
Timestamp Compiler: 2024-02-08
Update OS: Last Update OS (Ubuntu): Today, 11th of February 2024, around 11 a.m.
- OS -> Ubuntu 20.04.5 LTS (Linux x86-64)
- 64bits
CoApp: installed as sudo dpkg -i vdhcoapp-linux-x86_64.deb -> Newest version 2.0.17

Here the log from installing (sorry, some output is in my native language, german):

(Lese Datenbank ... 236041 Dateien und Verzeichnisse sind derzeit installiert.)
Vorbereitung zum Entpacken von vdhcoapp-linux-x86_64.deb ...
Uninstalling…
VdhCoApp : VdhCoApp has successfully unregistered itself.
Uninstalling…
Flatpak is installed. Making the coapp available from browser sandboxes:
Linked coapp within org.mozilla.firefox.
Linked coapp within com.brave.Browser.
Linked coapp within com.google.Chrome.
Linked coapp within com.google.ChromeDev.
Linked coapp within org.chromium.Chromium.
Linked coapp within com.github.Eloston.UngoogledChromium.
Linked coapp within com.microsoft.Edge.
Linked coapp within com.microsoft.EdgeDev.
Removing file /root/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
Removing file /root/.var/app/org.mozilla.firefox/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
VdhCoApp : VdhCoApp has successfully unregistered itself.
Entpacken von net.downloadhelper.coapp (2.0.17) über (2.0.10) ...
net.downloadhelper.coapp (2.0.17) wird eingerichtet ...
Installing…
VdhCoApp : VdhCoApp is ready to be used
Installing…
Flatpak is installed. Making the coapp available from browser sandboxes:
Linked coapp within org.mozilla.firefox.
Linked coapp within com.brave.Browser.
Linked coapp within com.google.Chrome.
Linked coapp within com.google.ChromeDev.
Linked coapp within org.chromium.Chromium.
Linked coapp within com.github.Eloston.UngoogledChromium.
Linked coapp within com.microsoft.Edge.
Linked coapp within com.microsoft.EdgeDev.
Writing /root/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
Writing /root/.var/app/org.mozilla.firefox/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
VdhCoApp : VdhCoApp is ready to be used
--
On Linux, the output of /opt/vdhcoapp/vdhcoapp --info ->
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/mac7pro",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e"
}

The message is the same: Coapp not installed -> "No such native application net.downloadhelper.coapp"- One click to check again does not help.
Any ideas? 

Thanks a lot in advance.

**Reply by paulrouget** [reply]
> Ubuntu 20.04.5

That's a old version of Ubuntu. Can you install Ubuntu 22?

**Reply by hector2o** [reply]
When I remember correctly 20.04.5 is still in update cycle and that version of ubuntu is up to date.
BTW on another PC where I run Ubuntu 22 I still have the same problem since Downloadhelper was updated at the beginning of this year. And on an older PC with an very old Windows 7 Version all works fine with downloadhelper and Coapp ;) (I know it is a not compareable setup). Any other advices do you have?

**Reply by paulrouget** [reply]
On your 20.04, update to 22.
For Ubuntu 22, please see the https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized - we have dealt with countless of similar issues, and the answer is always in that document.

**Comment by paulrouget** [comment, marked as answer]
@hector2o You're on Ubuntu. Make sure to update **Snap Core & Firefox** . There was a bug in Snap that got fixed a few days ago.

**Reply by hector2o** [reply]
See my answer to Grampa please, I did all I could do, but the coapp is still not recognised...

**Comment by hector2o** [comment]
The problem is solved. Downloadhelper, coapp and Mozilla released new versions. I updated ubuntu to 22.04 LTS and I installed now the coapp release 2.0.19  and tried it again and et voilá everythings works fine. Thanks a lot for your advices to that issue. I will close the discussion now.
